index.js

```diff
@@ -267,6 +269,8 @@ Router.prototype.find = function find (method, path) {
   var pindex = 0
   var params = []
   var i = 0
+  var previousPrefix = ''
+  var isStatic = true
```

Comment:
I think `isStatic`  should be reset to true at some point in the logic. If not we are not fixing this bug in its most complete form, as `/static/:parametric/static/:parametric` should be supported as well

Reply:
me too but the the following test passes with this implementation:
```js

  t.test('with parameter / 5', t => {
    t.plan(2)
    const findMyWay = FindMyWay()

    findMyWay.on('GET', '/bb/:foo/aa/', noop)
    findMyWay.on('GET', '/bb/:foo/aa/bulk', noop)

    t.equal(findMyWay.find('GET', '/bulk'), null)
    t.equal(findMyWay.find('GET', '/bb/foo/bulk'), null)
  })

```

Reply:
That's not the same as what @mcollina said. That test only has one `:parametric` part (`:foo`).

Reply:
I'm bit lost. The following test passes too:
```js

  t.test('with parameter / 6', t => {
    t.plan(3)
    const findMyWay = FindMyWay()

    findMyWay.on('GET', '/static/:parametric/static/:parametric', noop)
    findMyWay.on('GET', '/static/:parametric/static/:parametric/bulk', noop)

    t.equal(findMyWay.find('GET', '/bulk'), null)
    t.equal(findMyWay.find('GET', '/static/foo/bulk'), null)
    t.notEqual(findMyWay.find('GET', '/static/foo/static/bulk'), null)
  })

```

Let me know which test I should add

Reply:
the `isStatic` flag switches only once and it is not reset. I think it should be.

Reply:
Me too. But I don't know when and where. Without a failing test is very difficult for me to reset that flag.

Reply:
What does it mean `isStatic`?
The current node is static? The path is  static?

Reply:
All the node chain (node, its parent and so on) is static. This bug happens only on static route. I'm not able to reproduce it with the url with parameters